### PR TITLE
Fixes #5 code : handle pagination when saving user repositories

### LINKED ISSUE #5: code: handle pagination when saving user repositories
Github API uses pagination and lists only 30 items per page by default. But a user can have more than 30 repositories. Handle this situation in the `get_repositories` function defined in the [utils.py](https://github.com/ankan17/contrib/blob/master/core/utils.py) file. Use the `per_page` and the `page` options for traversing.

### PR DESCRIPTION
**Pull request by JayjeetAtGithub**

### CODE REVIEW

File: core/utils.py [hunk 1]
```diff
@@ -7,13 +7,24 @@
 from core.models import UserProfile, Repository
 
 
+"""Returns the request url with pagination"""
+def get_req_url(username,page):
```
Comment (ankan17) [inline core/utils.py]: This is used only once. No need of creating a separate function and thus introducing the function call overhead.

File: core/utils.py [hunk 2]
```diff
@@ -7,13 +7,24 @@
...
+    page = 1
     repos = []
-    if response.status_code == 200:
-        repos = response.json()
+    while True:
+    	response = requests.get(get_req_url(username,page))
+    	if len(response.json()) == 0:
+    		break
+    	if response.status_code == 200:
+    		repos = repos + response.json()
```
Comment (ankan17) [inline core/utils.py]: Use the pythonic way `repos.extend(response.json())`

File: core/utils.py [hunk 3]
```diff
@@ -7,13 +7,24 @@
...
     """Returns the repositories of a user using the Github API."""
-    repos_url = 'https://api.github.com/users/{}/repos'.format(username)
-    response = requests.get(repos_url)
+    page = 1
     repos = []
-    if response.status_code == 200:
-        repos = response.json()
+    while True:
+    	response = requests.get(get_req_url(username,page))
+    	if len(response.json()) == 0:
```
Comment (ankan17) [inline core/utils.py]: The status code checking should be done before this line.

File: core/utils.py [hunk 4]
```diff
@@ -7,13 +7,24 @@
...
+    		break
+    	if response.status_code == 200:
+    		repos = repos + response.json()
+    	page = page + 1
```
Comment (ankan17) [inline core/utils.py]: Use the shorthand `page += 1`

**Review by ankan17 — CHANGES_REQUESTED** [review]

File: core/utils.py (hunk 1 above)
Comment (JayjeetAtGithub) [reply core/utils.py]: i have used this function to get the modified url with different page number everytime
it is called more than once

Comment (ankan17) [reply core/utils.py]: This is called more than once, not used more than once. There is a difference. This line can be moved to the parent function itself.

File: core/utils.py (hunk 2 above)
Comment (JayjeetAtGithub) [reply core/utils.py]: done

File: core/utils.py (hunk 4 above)
Comment (JayjeetAtGithub) [reply core/utils.py]: Done

File: core/utils.py (hunk 3 above)
Comment (JayjeetAtGithub) [reply core/utils.py]: if response.status_code == 200:
    		if len(response.json()) == 0:
    		break
    		
    		repos.extend(response.json())
    	page += 1


should it be this ?

File: core/utils.py (hunk 1 above)
Comment (JayjeetAtGithub) [reply core/utils.py]: repos_url = 'https://api.github.com/users/{}/repos?page='.format(username)
response = requests.get(repos_url+str(page))

will this work ??

**Pull request closed without merging**